### PR TITLE
Split ISO metadata profile from csw modules and demote to community.

### PR DESCRIPTION
**Pull request by NielsCharlier**
As discussed on mailing list.

### CODE REVIEW

**Comment by aaime** [comment]
Build failed on a PMD QA check:

```
[INFO] --- maven-pmd-plugin:3.11.0:check (default) @ gs-csw-core ---
[INFO] PMD Failure: org.geoserver.csw.response.CSWObjectEncodingResponse:25 Rule:UnusedPrivateField Priority:3 Avoid unused private fields such as 'catalogStore'..
[INFO] PMD Failure: org.geoserver.csw.response.CSWObjectEncodingResponse:25 Rule:UnusedPrivateField Priority:3 Avoid unused private fields such as 'catalogStore'..
[INFO] FAILURE build of project org.geoserver.csw:gs-csw-core
```

In case you want to run the QA checks locally (slow, suggest to do so only on modified modules) you can build with ``-Dqa``, see also: https://docs.geoserver.org/latest/en/developer/qa-guide/index.html

**Comment by aaime** [comment]
@NielsCharlier did not forget about this one, just waiting GeoSolutions to give me working hours to look at it.

**Comment by NielsCharlier** [comment]
No problem, good to hear

**Comment by NielsCharlier** [comment]
No rush but any chance you can get us a timeframe? We now have conflicts and all those CSW issues are blocked by this + metadata community module which has a tutorial that depends on these changes

**Comment by aaime** [comment]
@NielsCharlier sorry no, I'm not in control of task schedule. You might want to send a private message to Simone.

File: src/community/csw-iso/src/test/java/org/geoserver/csw/store/internal/iso/GetDomainTest.java [hunk 1]
```diff
@@ -2,7 +2,7 @@
  * This code is licensed under the GPL 2.0 license, available at the root
  * application directory.
  */
-package org.geoserver.csw.records.iso;
```
[the remark this answers is not in the document]
Comment (NielsCharlier) [reply src/community/csw-iso/src/test/java/org/geoserver/csw/store/internal/iso/GetDomainTest.java]: With DC you must specify attribute dc:title
with the iso metadata profile, it's just "Title"
that is why there are two separate tests for each

File: src/community/release/ext-csw-iso.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    </formats>
+    <includeBaseDirectory>false</includeBaseDirectory>
+    <fileSets>
+        <fileSet>
+            <directory>release/target/dependency</directory>
+            <outputDirectory></outputDirectory>
+            <includes>
+                <include>gs-csw-iso*.jar</include>		
+            </includes>
+            <excludes>
```
[the remark this answers is not in the document]
Comment (NielsCharlier) [reply src/community/release/ext-csw-iso.xml]: to be honest... copy-pasted from ext-csw.xml. I have no idea.

**Comment by NielsCharlier** [comment]
changes added as requested

**Comment by NielsCharlier** [comment]
thanks @aaime

**Pull request merged**